src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -488,4 +494,124 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+            return this.utils.txInspectionHookEntities().stream().map(InspectionHookEntity::getHookId)
+                    .collect(Collectors.toList());
+        });
+    }
+
+    public Element updateInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
+        this.utils.throwExceptionIfNullElement(inspectionPortEntity);
+
+        if (inspectionPort.getIngressPort() == null) {
+            this.utils.throwExceptionIfNullElement(inspectionPort.getIngressPort(), "Ingress port is null");
+        }
+        if (inspectionPort.getEgressPort() == null) {
+            this.utils.throwExceptionIfNullElement(inspectionPort.getEgressPort(), "Egress port is null");
+        }
+
+        if (inspectionPort.getEgressPort().getElementId() == inspectionPort.getIngressPort().getElementId()) {
```

Comment:
use .equals comparison.

also load the ingress element from the db, and check if the egress element id provided is the same as the ingress entity id. this will eliminate a bit of code